framework/src/controller/channels/child_process_channel.js

```diff
@@ -137,9 +139,18 @@ class ChildProcessChannel extends BaseChannel {
 	 * @returns {Promise<void>}
 	 */
 	async cleanup() {
+		if (this.pubSocket && typeof this.pubSocket.close === 'function') {
```

Comment:
Why do we check for `this.pubSocket.close` function?

Reply:
To avoid `Uncaught TypeError`?

Reply:
if this.pubSocket is truthy that means it's already a socket object that has a close method. I'd prefer to not have that control for the sake of simplicity. Because person who writes the tests can think it's relevant with the logic etc.